resources/views/seasons/_rankings_table.blade.php

```diff
@@ -0,0 +1,79 @@
+{{--
+    Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+    See the LICENCE file in the repository root for full licence text.
+--}}
+
+<table class="ranking-page-table">
+    <thead>
+        <tr>
+            <th class="ranking-page-table__heading"></th>
+            <th class="ranking-page-table__heading ranking-page-table__heading--main"></th>
+            <th class="ranking-page-table__heading">
+                {{ osu_trans('rankings.stat.total_score') }}
+            </th>
+            <th class="ranking-page-table__heading ranking-page-table__heading--focused">
+                {{ osu_trans('rankings.stat.division') }}
+            </th>
+        </tr>
+    </thead>
+
+    <tbody>
+        @foreach ($scores as $index => $score)
+            @php
+                $rank = $scores->firstItem() + $index;
+
+                $currentThreshold = isset($currentDivision)
+                    ? $currentDivision['absolute_threshold']
+                    : 0;
+
+                if ($rank > $currentThreshold) {
+                    foreach ($divisions as $division) {
+                        if ($rank <= $division['absolute_threshold']) {
+                            $currentDivision = $division;
+                            break;
+                        }
+
+                        // this shouldn't ever happen if divisions are set up properly :)
+                        $currentDivision = null;
+                    }
+                }
+            @endphp
```

Comment:
less looping the divisions
```suggestion
        @foreach ($scores as $index => $score)
            @php
                $rank = $scores->firstItem() + $index;

                if (!isset($currentDivision) || $rank > $currentDivision['absolute_threshold']) {
                    while (($currentDivision = array_shift($divisions)) !== null) { {
                        if ($rank <= $currentDivision['absolute_threshold']) {
                            break;
                        }
                    }
                }
            @endphp
```